### PR TITLE
Handle deletion for message from outbox and header from inbox.

### PR DESCRIPTION
**Pull request by brian-js**
Requires https://github.com/maidsafe/mpid_messaging/pull/8 for mpid_messaging merged, version changed and published.

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/safe_vault/324)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @maqi

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/personas/mpid_manager.rs [hunk 1]
```diff
@@ -419,6 +416,68 @@ impl MpidManager {
...
+
+    pub fn handle_delete(&mut self, routing_node: &RoutingNode, request: &RequestMessage)
+            -> Result<(), InternalError> {
+        let (data, message_id) = match request.content {
+            RequestContent::Delete(Data::PlainData(ref data), ref message_id) => {
+                (data.clone(), message_id.clone())
+            }
+            _ => unreachable!("Error in vault demuxing"),
+        };
+        let mpid_message_wrapper = unwrap_option!(deserialise_wrapper(data.value()),
```
Comment (maqi) [inline src/personas/mpid_manager.rs]: the rfc defines the delete request not being wrapped in the MpidMessageWrapper.
which means the data is expected to be the MpidHeader or MpidMessage directly.

File: src/personas/mpid_manager.rs [hunk 2]
```diff
@@ -419,6 +416,68 @@ impl MpidManager {
...
+                if let Some(ref mut account) = self.accounts.get_mut(&request.dst.get_name().clone()) {
+                    let mut registered = false;
+
+                    if account.registered_clients().iter().any(|authority| *authority == request.src) {
+                        registered = true;
+                    }
+
+                    if let Ok(data) = self.chunk_store_outbox.get(&message_name) {
+                        if !registered {
+                            let mpid_message: MpidMessage = unwrap_result!(deserialise(&data));
```
Comment (maqi) [inline src/personas/mpid_manager.rs]: the current stored data in chunk_store, is a serialised MpidMessageWrapper::PutMessage(mpid_message).
refer https://github.com/maidsafe/safe_vault/blob/master/src/personas/mpid_manager.rs#L307-L312  for detail.

Comment (brian-js) [inline src/personas/mpid_manager.rs]: The code referred to is for the MpidMessageWrapper::PutHeader and should probably be updated to store the serialised Header.

**Pull request merged**